packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +61,106 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const isValidDate = (date: Date) => date instanceof Date && !isNaN(date.valueOf());
+
+const calculateTimezoneByValueCustom = (timezone: string, date = new Date()) => {
+  const customTimezones = timeZoneDataUtils.getTimeZones();
+  if (customTimezones.length === 0) {
+    return undefined;
+  }
+
+  const dateUtc = createUTCDate(date);
+  return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+};
+
+const calculateTimezoneByValue = (timeZone: string | undefined, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+  const isValidTimezone = timeZoneList.value.includes(timeZone);
+  if (!isValidTimezone) {
+    errors.log('W0009', timeZone);
+    return undefined;
+  }
+
+  if (!isValidDate(date)) {
+    return undefined;
+  }
+
+  let result = calculateTimezoneByValueCustom(timeZone, date);
+  if (result === undefined) {
+    result = calculateTimezoneByValueCore(timeZone, date);
+  }
+  return result;
+};
+
+const calculateTimezoneByValueCore = (timeZone: string, date = new Date()): number | undefined => {
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    errors.log('W0009', timeZone);
+    return undefined;
```

Comment:
Are you sure the error needs to be handled? Maybe an error should be thrown and the code execution should be interrupted?